src/gleam/float.gleam

```diff
@@ -267,6 +267,31 @@ pub fn truncate(x: Float) -> Int {
 @external(javascript, "../gleam_stdlib.mjs", "truncate")
 fn do_truncate(a: Float) -> Int
 
+/// Converts the value to a given precision as a `Float`.
+/// The precision is the number of allowed decimal places.
+/// Negative precisions are allowed and force rounding
+/// to the nearest tenth, hundredth, thousandth etc.
+///
+/// ## Examples
+///
+/// ```gleam
+/// to_precision(2.43434348473, precision: 2)
+/// // -> 2.43
+/// ```
+///
+/// ```gleam
+/// to_precision(547890.453444, precision: -3)
+/// // -> 548000.0
+/// ```
+///
+pub fn to_precision(x: Float, precision: Int) -> Float {
+  do_to_precision(x, precision)
```

Comment:
Can we not implement this in Gleam?

Reply:
@lpil It can be done in gleam but it's around 64% faster for Erlang and  2390% faster for JS to do it with the FFI setup compared to pure gleam so I think the performance is worth the FFI

Test done over 1 billion iterations of trimming a random float to precision 4, duration in milliseconds

`Erlang`
```sh
Done
Duration(49189853) # FFI
Done
Duration(81002269) # Gleam
```

`Javascript`
```sh
Done
Duration(6086723) # FFI
Done
Duration(145866758) # Gleam
```

Source for the test

```gleam
import birl
import gleam/float
import gleam/int
import gleam/io
import gleam/result

pub fn main() {
  let ffi_start = birl.now()
  to_precision_ffi_test(1_000_000_000)
  io.debug(birl.difference(birl.now(), ffi_start))
  let gleam_start = birl.now()
  to_precision_gleam_test(1_000_000_000)
  io.debug(birl.difference(birl.now(), gleam_start))
}

fn to_precision_ffi_test(times: Int) {
  float.random()
  |> float.to_precision(4)

  case times < 0 {
    True -> io.println("Done")
    False -> to_precision_ffi_test(times - 1)
  }
}

fn to_precision_gleam_test(times: Int) {
  float.random() |> gleam_to_precision(4)

  case times < 0 {
    True -> io.println("Done")
    False -> to_precision_gleam_test(times - 1)
  }
}

fn gleam_to_precision(value: Float, precision: Int) -> Float {
  let factor =
    float.power(10.0, -precision |> int.to_float) |> result.unwrap(0.0)
  float.round(value |> float.divide(factor) |> result.unwrap(0.0))
  |> int.to_float
  |> float.multiply(factor)
}
```